source/containerimage/pull.go

```diff
@@ -39,13 +42,21 @@ import (
 // TODO: break apart containerd specifics like contentstore so the resolver
 // code can be used with any implementation
 
+type ContainerImageResolverType int
+
+const (
+	ResolverTypeRegistry ContainerImageResolverType = iota
+	ResolverTypeOCILayout
+)
+
 type SourceOpt struct {
 	Snapshotter   snapshot.Snapshotter
 	ContentStore  content.Store
 	Applier       diff.Applier
 	CacheAccessor cache.Accessor
 	ImageStore    images.Store // optional
 	RegistryHosts docker.RegistryHosts
+	Source        ContainerImageResolverType
```

Comment:
This should be `ResolverType`

Reply:
On the client side as well https://github.com/moby/buildkit/pull/2827#discussion_r897541477

Reply:
Sorry, I meant the "Source" key should be changed (changing the type is fine as well). Both on pull.go and llb/resolver.go

Reply:
I don't understand what you mean by "Source key should be changed"?

Oh! You mean calling it `SourceOpt.ResolverType` instead of `SourceOpt.Source`? Sure, I can do that.

Reply:
Done. Had to change all of the dependencies, but done. Let's let CI work it out, unless you gave other changes.